base_recurrence/models/recurrence_mixin.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 ACSONE SA/NV
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class RecurrenceMixin(models.AbstractModel):
+
+    _name = "recurrence.mixin"
+    _description = "Recurrence Mixin"
+    _field_last_recurrency_date = None
+    _field_next_recurrency_date = None
+
+    # Scheduling
+    recurrence_type = fields.Selection(
+        [
+            ("daily", "Daily"),
+            ("weekly", "Weekly"),
+            ("monthly", "Monthly"),
+            ("monthlylastday", "Monthly Last Day"),
+            ("quarterly", "Quarterly"),
+            ("semesterly", "Semesterly"),
+            ("yearly", "Yearly"),
+        ]
+    )
+
+    @api.model
+    def get_relative_delta(self, recurring_rule_type, interval=1):
+        """Return a relativedelta for one period.
+
+        When added to the first day of the period,
+        it gives the first day of the next period.
+        """
+        recurring_func = "_recurring_rule_type_" + recurring_rule_type
+        if hasattr(self, recurring_func):
+            return getattr(self, recurring_func)(interval=interval)
+        else:
+            raise NotImplementedError()
+
+    # HELPER FUNCTIONS #
```

Comment:
I think need add `api.model` decorator for all HELPER FUNCTIONS witch start with `_recurring`.

Reply:
that is true! I missed it in my review.

Reply:
ok